core/lib/spree/core/controller_helpers/auth.rb

```diff
@@ -24,9 +24,9 @@ def redirect_back_or_default(default)
           session['spree_user_return_to'] = nil
         end
 
-        def set_guest_token
-          if cookies.signed[:guest_token].blank?
-            cookies.permanent.signed[:guest_token] = { value: generate_guest_token, httponly: true }
+        def set_token
```

Comment:
@kushniryb @bbonislawski I think we shouldn't change the cookie now - this would be really dangerous

Reply:
@damianlegawiec Why do you think that'll be dangerous?

Reply:
@kushniryb all of the issued cookies will stop working, we cannot change the cookie name for now

Reply:
What about adding something like this?
```
if cookies.signed[:guest_token].present?
  cookies.permament.signed[:token] ||= cookies.signed[:guest_token]
end
```

Reply:
We could clean `guest_token` with this too so end user won't feel any change and everyone will migrate in short time

Reply:
@bbonislawski that's a very good idea, let's set the new cookie and delete the new one, and after 4.0 we'll remove this code